doc/advanced_configuration.rst

```diff
@@ -142,6 +143,29 @@ If you so desire you can implement your own sorting key. It will be
 provided the relative paths to `conf.py` of each sub gallery folder.
 
 
+.. _within_gallery_order:
+
+Sorting gallery examples
+========================
+
+Within a given gallery (sub)section, the example files by default are ordered
+by the amount of code, i.e. :class:`sphinx_gallery.sorting.AmountOfCodeSortKey`
+as::
+
+    from sphinx_gallery.sorting import AmountOfCodeSortKey
+    sphinx_gallery_conf = {
+        ...
+        'within_subsection_order': AmountOfCodeSortKey,
+    }
+
+However, other options exist that can be instantiated for use with
+``within_subsection_order``:
+
+- :class:`sphinx_gallery.sorting.FileSizeSortKey` to sort by file size.
+- :class:`sphinx_gallery.sorting.FileNameSortKey` to sort by file name.
```

Comment:
just a thought on `ExplicitFilenameSortKey` (which I know you aren't implementing here): I think it would get really confusing if we implemented this, since right now doing explicit sorting for folders *requres* that all folders are listed. We couldn't do the same thing for explicit filenames because there would be too many IMO